packages/angular-workspace/nimble-angular/table/testing/table.pageobject.ts

```diff
@@ -1,3 +1,35 @@
-import { TablePageObject } from '@ni/nimble-components/dist/esm/table/testing/table.pageobject';
+import { TablePageObject as NimbleComponentsTablePageObject } from '@ni/nimble-components/dist/esm/table/testing/table.pageobject';
+import { waitForUpdatesAsync } from '@ni/nimble-angular';
+import type { Table, TableRecord } from '@ni/nimble-angular/table';
 
-export { TablePageObject };
\ No newline at end of file
+/**
+ * The page object for the `nimble-table` component to provide consistent ways of querying
+ * and interacting with the component during tests.
+ *
+ * This Angular version of the page object extends the nimble-components version to add the ability
+ * to wait for data updates to be applied to the table since the timing isn't easily observable
+ * when using the `data$` property.
+ */
+export class TablePageObject<T extends TableRecord> extends NimbleComponentsTablePageObject<T> {
```

Comment:
>🧪 Testing
>
>None. Both the implementation of getHeaderTextContent and waitForDataUpdatesToRender were copied from the sl-table page object in SystemLinkShared.

Even when a page object method isn't needed by Nimble component tests we've still tried to write dedicated specs to cover them. It can be pretty basic, just a single test that verifies the method is sane, but it is helpful to catch breakages in Nimble's tests rather than client tests. I think we should still strive for this for these two methods.

 [This file shows the pattern for the table in nimble-components](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/tests/table-pageobject.spec.ts) which you could copy for the nimble-angular page object and add a test to for the nimble-components page object.

Reply:
Wrote tests and updated PR description.

Reply:
Based on Jesse's comment I think I'd also expect tests in the nimble-components `table-pageobject.spec.ts` for the new `getHeaderTextContent`. But I'd also be curious if that's our general policy now for page object additions (not something I've been looking for explicitly in review).

Reply:
I updated an existing test to use the new page object function, so I thought that would be sufficient.

https://github.com/ni/nimble/pull/2308/commits/32ddbbfc20b33d9216ddd18fa160bc330b744adf

Reply:
(making up on the spot) I think since we are making pageobjects public apis they probably warrant their own explicit tests. Future refactoring could remove the current coverage of the pageobject but clients may still rely on the behavior if it's not removed from the public api explicitly. I think I could be convinced either way, I don't think we have page object apis yet that are only used externally but those absolutely need coverage.

Reply:
To date I have only enforced that all page object methods should get called somewhere by some Nimble test, not that they get called by a dedicated page object test. For this PR I'd advocate to hold the same bar, which with the new tests I believe is now being met.

I think your proposal likely makes sense as a policy going forward. It seems like it'd be easy for us to remove the only usage of a page object method and not notice that it's no longer tested. But there are likely details to work out so we should split the discussion into a tech debt issue. I'll update the notes on #265 and mark it for triage.